### PR TITLE
fix: ensure aria-hidden & inert is handled correctly with open modals in different carbon instances

### PR DESCRIPTION
**Pull request by tomdavies73**
### Proposed behaviour

Adds an extra step of validation in `useModalAria` with the addition of a `data-modal-hidden` attribute to ensure multiple instances open at the same time only take ownership of managing elements it encounters. Stops separate instances of the hook restoring elements or marking elements (adding attributes) which they should not have any control over, and places the hook into a queue, with the next hook only managing elements after the previous one has finished.

### Current behaviour

Currently, we use a strong method of marking all elements which are not in the root toward the modal as `aria-hidden` and `inert`, we then clean up by removing all the attributes we previously added. This is fine in standard use cases, but causes issues when we have two open stacked modals running in two separate carbon instances, as they can incorrectly restore or add attributes, which causes a bug in production where everything remains `aria-hidden` and `inert` even when all modals are closed.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Playwright automation tests added or updated if required
- [x] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

I have added the `"Separate CarbonInstances With Dialog"` Adaptive Sidebar test story to replicate an edge case where two instances of Carbon are used at the same time. 

**Steps to reproduce are:**

Open the adaptive sidebar on a large screen size, open the dialog, then re-size so the adaptive sidebar becomes a standard sidebar, then close the sidebar & close the dialog after. You can still interact with all other elements and even repeat the steps again if you wish. 

However, before the fix using this reproduction and repeating the same steps, you can see that this is not possible. The rest of the page remains `aria-hidden` and `inert`: 

https://stackblitz.com/edit/parsium-carbon-starter-a3u4txxq?file=src%2FApp.tsx%3AL22-L22,src%2Fmain.tsx,index.html

### CODE REVIEW

File: src/components/adaptive-sidebar/adaptive-sidebar-test.stories.tsx [hunk 1]
```diff
@@ -469,6 +472,73 @@ ExampleImplementation.parameters = {
...
+            <Instance2 />
+          </CarbonProvider>
+        </>,
+      );
+    }
+
+    return () => {
+      if (container2) {
+      }
+    };
```
Comment (damienrobson-sage) [inline src/components/adaptive-sidebar/adaptive-sidebar-test.stories.tsx]: **question**: why is this useEffect's clean-up empty?

Comment (Parsium) [inline src/components/adaptive-sidebar/adaptive-sidebar-test.stories.tsx]: **issue**: this effect initialises the React root but doesn't destroy it afterwards. Adding a cleanup would ensure we don't leave root in memory when the component unmounts:

```suggestion
    let root = Root | null = null;

    const container = document.getElementById("mfe-2-root");
    if (container && !container.hasChildNodes()) {
      root = createRoot(container);
      root.render(
        <CarbonProvider>
          <Instance2 />
        </CarbonProvider>
      );
    }

    return () => {
      root?.unmount();
      root = null;
    };
```
